packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.tsx

```diff
@@ -0,0 +1,277 @@
+
+  const {
+    isOpen,
+    selectItem: selectedItemCombobox,
+    getMenuProps,
+    getInputProps,
+    getToggleButtonProps,
+    highlightedIndex,
+    getItemProps,
+  } = useCombobox({
+    items: filteredItems,
+    selectedItem: null,
+    ...(itemToString && { itemToString }),
+    inputValue,
+    stateReducer: (_state, actionAndChanges) => {
+      const { changes, type } = actionAndChanges;
+      switch (type) {
+        case useCombobox.stateChangeTypes.InputKeyDownEnter:
+        case useCombobox.stateChangeTypes.ItemClick:
+          return { ...changes, isOpen: true };
+        default:
+          return changes;
+      }
+    },
+    onStateChange({ inputValue: newInputValue, type, selectedItem: newSelectedItem }) {
+      switch (type) {
+        case useCombobox.stateChangeTypes.InputKeyDownEnter:
+        case useCombobox.stateChangeTypes.ItemClick:
+        case useCombobox.stateChangeTypes.InputBlur:
+          if (newSelectedItem) {
+            addSelectedItem(newSelectedItem);
+          }
+          setInputValue('');
+          break;
+        case useCombobox.stateChangeTypes.InputChange:
+          setInputValue(newInputValue);
+          break;
+        default:
+          break;
+      }
+    },
+    ...useComboboxProps,
+  });
+
+  return (
+    <AutocompleteMultiSelectProvider
+      value={{
+        isOpen,
+        size,
+        isDisabled,
+        items: filteredItems,
+        itemToString,
+        getMenuProps,
+        getInputProps,
+        getItemProps,
+        getToggleButtonProps,
+        selectedItem: selectedItemCombobox,
+        highlightedIndex,
+        selectedItems,
+        removeSelectedItem,
+        getSelectedItemProps,
+        getDropdownProps,
+        reset: () => setInputValue(''),
+      }}
+    >
+      <Box position="relative">{children}</Box>
+    </AutocompleteMultiSelectProvider>
+  );
+};
+
+const AutocompleteMultiSelectInput: FC<AutocompleteMultiSelectInputProps> = ({
+  clearButtonMode = 'has-value',
+  size,
+  isDisabled,
+  ...rest
+}) => {
+  const {
+    reset,
+    getInputProps,
+    getToggleButtonProps,
+    getDropdownProps,
+    isOpen,
+    size: autocompleteMultiselectSize,
+    isDisabled: autocompleteMultiSelectDisabled,
+  } = useAutocompleteMultiSelectContext();
+  const ref = useRef<HTMLInputElement>(null);
+  const inputProps = getInputProps({ ...getDropdownProps({ preventKeyAction: isOpen, ref }) });
+
+  const clearButtonConditions = {
+    'has-value': inputProps.value,
+    never: false,
+    always: true,
+  };
+  const shouldShowClearButton = clearButtonConditions[clearButtonMode];
+  const inputSize = size ?? autocompleteMultiselectSize;
+  const isInputDisabled = isDisabled || autocompleteMultiSelectDisabled;
+
+  const onReset = useCallback(() => {
+    reset?.();
+    ref?.current?.focus?.();
+  }, []);
+
+  const pointerEventsStyle: {} = isInputDisabled ? { pointerEvents: 'none' } : {};
```

Comment:
nit: is the type definition as `{}` correct/needed?

Reply:
It was not really needed. Thanks!

Reply:
Updated to a more precise type. Thanks!